mentii/mentii
# Change step

This change allows the user to select the root of the badstep "subtree" and get mentii back on track. Only the step where the problem actually occurred can be corrected.

**Significant Changes**
1. ng-bootstrap has been included so that we can use the javascript side of bootstrap within angular such as presenting modal popups.

**How to Test**
1. Log in
2. Go to /problem/display/d/a[1,2,3]
3. Step through the problem until you get to a bad step
4. Go to the next bad step and click "This Step is Incorrect". You should see an alert that yes the step is part of a larger problem, but its not where it should be corrected.
5. Click on the "This Step is Incorrect" on the root bad step and you should be presented with a modal.
6. Enter the correction for what mentii should have done. **This is _exact_ string matching so I suggest copying the problem from the description of the modal and putting it in the text box, then make your changes.**
7. If the correction is correct, the modal will close and mentii will progress to the next step. Otherwise an alert will tell you that the correction was still wrong.

**Notes**
🐑   The bad steps are still red for your convenience.
